### PR TITLE
minor improvments to SWATHWorkflow

### PR DESCRIPTION
**Pull request by cbielow**
- improve docs
- optimize loop end condition
- avoid new/delete
- use platform specific temp dir (which also changes with environment automatically)
- force directory separator at the end of tempDir to avoid late crash due to file-not-found for internal temp files

### CODE REVIEW

**Comment by hroest** [comment]
looks good, it seems like travis has some issues now: 

```
Error: Unable to write file (the file '/home/travis/build/OpenMS/OpenMS/_build/src/tests/topp/osw_file5/openswath_tmpfile_ms1.mzML' could not be created)
```

**Comment by hroest** [comment]
> use platform specific temp dir (which also changes with environment automatically)

I dont think that quite worked as expected, as now it tries to write the file `osw_file5/openswath_tmpfile_ms1` instead of the intended `osw_file5_openswath_tmpfile_ms1` file. @cbielow you will have to change the CMake test file, but also add an additional dependency to make sure the temp files are not overwritten as they will have the same name ... 

Currently it was possible to specify filename and directory, but now this is not possible any more, leading to the test failures.

File: src/utils/OpenSwathWorkflow.cpp [hunk 1]
```diff
@@ -682,7 +684,10 @@ class TOPPOpenSwathWorkflow
 
     String readoptions = getStringOption_("readOptions");
     String mz_correction_function = getStringOption_("mz_correction_function");
-    String tmp = getStringOption_("tempDirectory");
+    
+    // make sure tmp is a directory with proper separator at the end (downstream methods simply do path + filename)
+    // (do not use QDir::separator(), since its platform specific (/ or \) while absolutePath() will always use '/')
+    String tmp_dir = String(QDir(getStringOption_("tempDirectory").c_str()).absolutePath()).ensureLastChar('/');
```
Comment (hroest) [inline src/utils/OpenSwathWorkflow.cpp]: before it was possible to specify filename and directory, but now this is not possible any more, leading to the test failures. This was a good idea if you wanted to analyze multiple runs without worrying about them overwriting each others files.

**Comment by cbielow** [comment]
ok, I see the problem, but IMHO this was misleading. `tempDirectory` was not a directory, but rather a prefix to a fixed suffix.
The proper fix would be to use proper random filenames in the specified directory. This would avoid to specifiy artificial dependencies in ctest as well. Does that give headaches to anyone? If desired, one could print the filename to the console, in case someone wants to check the content (but the cache file is binary, so its hard to inspect anyway).

**Comment by hroest** [comment]
> The proper fix would be to use proper random filenames in the specified directory.

:+1: 

I think right now the test 5 and 6 will overwrite each others files ...

**Comment by cbielow** [comment]
ready for merge

**Comment by oliveralka** [comment]
+1 
macOS fail is unrelated

**Pull request merged**